meta.go

```diff
@@ -266,19 +272,27 @@ func setMetaValueRecursive(key string, value string, previousMeta interface{}) (
 			var tmpValue interface{}
 			previousMetaMap := convertInterfaceToMap(previousMeta)
 			if previousMetaMap[keyHead] == nil {
-				childKey, tmpValue = setMetaValueRecursive(childKey, value, previousMetaMap)
+				childKey, tmpValue = setMetaValueRecursive(childKey, value, previousMetaMap, jsonValue)
 			} else {
 				// copy previous object only if it is map
 				previousObj := convertInterfaceToMap(previousMetaMap[keyHead])
 				if len(previousObj) != 0 {
 					obj = previousObj
 				}
-				childKey, tmpValue = setMetaValueRecursive(childKey, value, previousMetaMap[keyHead])
+				childKey, tmpValue = setMetaValueRecursive(childKey, value, previousMetaMap[keyHead], jsonValue)
 			}
 			obj[childKey] = tmpValue
 			return keyHead, obj
 		}
 	}
+	if jsonValue {
+		var objectValue interface{}
+		if err := json.Unmarshal([]byte(value), &objectValue); err == nil {
+			return key, objectValue
+		} else {
+			log.Panic(err)
```

Comment:
I don't know well which should we choose `log.Panic` or `log.Fatal`.  Do we need to raise `panic` in this case?

Reply:
Well… in general, I prefer panic - it _can_ be caught, but in practice, it just results in fatal with a stack-trace which can prove helpful in debugging.  I could be wrong, but I think Fatal just exits with no facility to "catch" the panic.

Reply:
make sense. thanks!